public/app/features/browse-dashboards/state/reducers.ts

```diff
@@ -200,3 +201,19 @@ export function setAllSelection(
     }
   }
 }
+
+type UIDLike = string | undefined;
+export function clearFolders(state: BrowseDashboardsState, action: PayloadAction<UIDLike | UIDLike[]>) {
```

Comment:
do we need to handle the singular case? in `onRestore` we only ever call it with an array, think it would be simpler to just make it `PayloadAction<UIDLike[]>`